app/service/registration.js

```diff
@@ -24,38 +25,32 @@ class RegistrationService {
   }
 
   /**
-   * @type {RegistrationService.GenerateOneTimeCode}
+   * @type {RegistrationService.getUserFromRequest}
    */
-  generateOneTimeCode () {
-    return `${(Math.floor(100000 + randomInt(900000)))}`
+  getUserFromRequest (request) {
+    const payload = request.auth?.artifacts?.decoded?.payload
+    if (payload) {
+      const { username } = payload
+      return username
+    }
+    throw new NotFoundError('user not found')
   }
```

Comment:
Just a thought on this...found when changes were done on my branch that calling `request.auth?.artifacts?.decoded?.payload` means that any changes will have to be duplicated.  If we use getCallingUser it will make it easy to do changes in future.

Reply:
I've added a new fn on [6f7353e](https://github.com/DEFRA/aphw-ddi-api/pull/258/commits/6f7353e9787c504c625aa20de38b574cf35e632b) - can be cherry picked.